File: packages/capacitor-actito-assets/ios/Plugin/ActitoAssetsPlugin.swift
import ActitoAssetsKit
import ActitoKit
import Capacitor
import Foundation

@objc(ActitoAssetsPlugin)
public class ActitoAssetsPlugin: CAPPlugin {
    @objc func fetch(_ call: CAPPluginCall) {
        guard let group = call.getString("group") else {
            call.reject("Missing 'group' parameter.")
            return
        }

        Actito.shared.assets().fetch(group: group) { result in
            switch result {
            case let .success(assets):
                do {
                    call.resolve([
                        "result": try assets.map { try $0.toJson() }
                    ])
                } catch {
                    call.reject(error.localizedDescription)
                }
            case let .failure(error):
                call.reject(error.localizedDescription)
            }
        }
    }
}

File: packages/capacitor-actito-assets/android/src/main/java/com/actito/assets/capacitor/ActitoAssetsPlugin.kt
package com.actito.assets.capacitor

import com.actito.Actito
import com.actito.ActitoCallback
import com.actito.assets.ktx.assets
import com.actito.assets.models.ActitoAsset
import com.getcapacitor.JSArray
import com.getcapacitor.JSObject
import com.getcapacitor.Plugin
import com.getcapacitor.PluginCall
import com.getcapacitor.PluginMethod
import com.getcapacitor.annotation.CapacitorPlugin

@CapacitorPlugin(name = "ActitoAssetsPlugin")
public class ActitoAssetsPlugin : Plugin() {
    @PluginMethod
    public fun fetch(call: PluginCall) {
        val group = call.getString("group") ?: run {
            call.reject("Missing 'group' parameter.")
            return
        }

        Actito.assets().fetch(group, object : ActitoCallback<List<ActitoAsset>> {
            override fun onSuccess(result: List<ActitoAsset>) {
                try {
                    call.resolve(
                        JSObject().apply {
                            put("result", JSArray().apply {
                                result.forEach { put(it.toJson()) }
                            })
                        }
                    )
                } catch (e: Exception) {
                    call.reject(e.localizedMessage)
                }
            }

            override fun onFailure(e: Exception) {
                call.reject(e.localizedMessage)
            }
        })
    }
}
